src/components/IPFSSubpathRedirect.tsx

```diff
@@ -0,0 +1,34 @@
+import { CID } from 'multiformats'
+import { useEffect } from 'react'
+
+export const IPFSSubpathRedirect = () => {
+  useEffect(() => {
+    const hashFromBase = extractIPFSHashFromBasePath()
+
+    if (!hashFromBase) {
+      console.info('Base ref does not contain an IPFS hash')
+      return
+    }
+
+    const redirectUrl = generateIpfsSubdomainUrl(hashFromBase)
+    if (!redirectUrl) return
+
+    // redirect to IPFS subdomain url
+    window.location.href = redirectUrl
+  }, [])
+
+  return null
+}
+
+function extractIPFSHashFromBasePath() {
+  const htmlBase = document.querySelector('base')
+  if (!htmlBase) return
+  const cidRegex =
+    /\/ipfs\/(Qm[1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})/
```

Comment:
```suggestion
  const cidRegex = /\/ipfs\/(Qm[1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})/
```
Looks like you had one line originally but changed it.  Why?

Reply:
Uniswap linter splits too long lines. I haven't touched the linter rules